gitlabbot/gitlabbot/handler.go

```diff
@@ -0,0 +1,253 @@
+	return &Handler{
+		DebugOutput: base.NewDebugOutput("Handler", kbc),
+		kbc:         kbc,
+		db:          db,
+		requests:    requests,
+		config:      config,
+		httpPrefix:  httpPrefix,
+		secret:      secret,
+	}
+}
+
+func (h *Handler) HandleNewConv(conv chat1.ConvSummary) error {
+	welcomeMsg := "Hi! I can notify you whenever something happens on a GitLab repository. To get started, set up a repository by sending `!gitlab subscribe <username/repo>`"
+	return base.HandleNewTeam(h.DebugOutput, h.kbc, conv, welcomeMsg)
+}
+
+func (h *Handler) HandleAuth(msg chat1.MsgSummary, _ string) error {
+	return h.HandleCommand(msg)
+}
+
+func (h *Handler) HandleCommand(msg chat1.MsgSummary) error {
+	if msg.Content.Text == nil {
+		h.Debug("skipping non-text message")
+		return nil
+	}
+
+	cmd := strings.ToLower(strings.TrimSpace(msg.Content.Text.Body))
+	if !strings.HasPrefix(cmd, "!gitlab") {
+		h.Debug("ignoring non-command message")
+		return nil
+	}
+
+	identifier := base.IdentifierFromMsg(msg)
+	tc, err := base.GetOAuthClient(identifier, msg, h.kbc, h.requests, h.config, h.db,
+		base.GetOAuthOpts{
+			AuthMessageTemplate: "Visit %s\n to authorize me to set up GitLab notifications.",
+		})
+	if err != nil || tc == nil {
+		return err
+	}
+
+	token, err := h.db.GetToken(identifier)
+	if err != nil {
+		h.Debug("error getting token from db")
+		return err
+	}
+
+	client := gitlab.NewOAuthClient(tc, token.AccessToken)
+
+	switch {
+	case strings.HasPrefix(cmd, "!gitlab subscribe"):
+		return h.handleSubscribe(cmd, msg, true, client)
+	case strings.HasPrefix(cmd, "!gitlab unsubscribe"):
+		return h.handleSubscribe(cmd, msg, false, client)
+	}
+	return nil
+}
+
+func (h *Handler) handleSubscribe(cmd string, msg chat1.MsgSummary, create bool, client *gitlab.Client) (err error) {
+	toks, err := shellquote.Split(cmd)
+	if err != nil {
+		return fmt.Errorf("error splitting command: %s", err)
+	}
+	args := toks[2:]
+	if len(args) < 1 {
+		return fmt.Errorf("bad args for subscribe: %v", args)
+	}
+
+	// Check if command is subscribing to a branch
+	if len(toks) == 4 {
+		return h.handleSubscribeToBranch(cmd, msg, create, client)
+	}
+
+
+	var message string
+	defer func() {
+		if message != "" {
+			_, err = h.kbc.SendMessageByConvID(msg.ConvID, fmt.Sprintf(message, args[0]))
+			if err != nil {
+				err = fmt.Errorf("error sending message: %s", err)
+			}
+		}
+	}()
+
+	alreadyExists, err := h.db.GetSubscriptionForRepoExists(msg.ConvID, args[0])
+	if err != nil {
+		return fmt.Errorf("error checking subscription: %s", err)
+	}
+
+	parsedRepo := strings.Split(args[0], "/")
+	if len(parsedRepo) != 2 {
+		return fmt.Errorf("invalid repo: %s", args[0])
+	}
+	if create {
+		if !alreadyExists {
+			project, err := getProject(args[0], client)
+			if err != nil {
+				return fmt.Errorf("error getting gitlab project: %s", err)
+			}
+
+			var webhookURL = h.httpPrefix + "/gitlabbot/webhook"
```

Comment:
`var` syntax is usually reserved for variable declaration `var foo string` why not use `:=` here?